site/content/blog/2022-08-22-the-param-miner-addon/index.md

```diff
@@ -0,0 +1,49 @@
+
+This post highlights the available features of the Param-Miner add-on (as of 2022-08-22).
+
+## How to use the Param-Miner add-on
+To begin with, you will need install it from the [Marketplace](/addons/) from within ZAP as it's not installed by default. Once you have it installed, you can access the Param Miner from three locations: the add-on menu which appears after clicking the plus green icon, the right click menu under the Attack option and from the tools menu. 
+![Right click menu](images/right-click-menu-example.png)
+
+This is how the Param Miner panel in the information window looks like:
+![Param Miner Tab](images/paramminer.png)
+
+Whether you choose to start a new scan by clicking the "New Scan" button on the Param Miner tab or start a new scan from the attack menu or the tools menu, you will be greeted with the Param Miner dialog. This dialog allows you to select the parameters you want to discover, the methods you want to use, the word lists you want to use and other options.  
+![Param Miner Dialog](images/param-miner-dialog.png)
+
+As of 2022-08-20 the Param Miner only supports the "URL Guess", which uses a list of URL parameters. So it is the one which is selected by default. The screenshot of the default "URL Guess" tab is given below. You can use the predefined wordlist provided by the add-on or you can create your own and import it after selecting the "Custom" option from the Wordlist pull down. Selecting "Custom" would also enable the File Location field for you where you can supply the appropriate file path after browsing. You can also use both the wordlists provided by the add-on and your own by selecting the "Both" option. 
+
+
+After selection of the word list, you have to select the methods. By default it's GET. You can select as many methods as you want using Ctrl + Click (For Mac or other related platforms use Cmd + Click). 
+
+![URL Guess Tab](images/url-guess-tab.png)
+
+
+## Future Work
+
+While the foundation of the add-on has been laid out, the Param Miner is still in its early stage. There is a lot more work left to do. I would be adding more functionality such as a starter string for JSON method in URL Guesser, adding a new method for XML and other guessing methods in the Param Miner dialog. I would also be adding "Parameter Mining" using "FAT GET" requests and other techniques as suggested in [Practical Web Cache Posinoning](https://portswigger.net/research/practical-web-cache-poisoning) and [Web Cache Entanglement](https://portswigger.net/research/web-cache-entanglement) which can be enabled or disabled based on users' choice, having the name "web cache poisoning" or other such related term.
+
+I would also work on add-on's integration with the Automation Framework in ZAP. This would help users to find parameters on every URLs in the website by using the Spider and launch active or passive scans against the results.
```

Comment:
```suggestion
I would also like to work on the add-on's integration with the ZAP Automation Framework. This would help users to find parameters on every URL in the website by using the Spider and launch active or passive scans against the results.
```